src/containers/Home/Restaurant.js

```diff
@@ -0,0 +1,229 @@
+// @flow
+import React from 'react';
+import {
+  View,
+  Modal,
+  Image,
+  ToastAndroid,
+} from 'react-native';
+import Config from 'react-native-config';
+import querystring from 'query-string';
+import { connect } from 'react-redux';
+import Icon from 'react-native-vector-icons/MaterialIcons';
+import FloatingActionButton from '../../components/FloatingActionButton';
+import colors from '../../themes/color';
+import styles from './style';
+import notfound from '../../images/notfound.png';
+import { actionCreators as placeActionCreators } from '../../redux/modules/place';
+import { actionCreators as favouriteActionCreators } from '../../redux/modules/favourite';
+
+type Place = {
+  latitude: number;
+  longitude: number;
+  name: string;
+  opening_hours: ?{ open_now: boolean };
+  permanently_closed: ?boolean;
+  place_id: string;
+  rating: ?number;
+  vicinity: string;
+  geometry: Object;
+};
+
+type State = {
+  photos: String[],
+  listFound: boolean,
+}
+
+type Props = {
+ doSetRes: Function,
+ doDumpPhoto: Function,
+ doAddNewPlace: Function,
+ doCreateNewList: Function,
+ visibility: boolean,
+ photo: String[],
+ place: Place,
+ listTitle: string,
+ favouriteList: Object[],
+}
+
+const PLACES_PHOTO_API = 'https://maps.googleapis.com/maps/api/place/photo?';
+const key = Config.GOOGLE_MAPS_API_KEY;
+
+class Restaurant extends React.Component<Props, State> {
+  static props = {
+    visibility: false,
+    photo: [],
+    place: null,
+    listTitle: '',
+  };
+
+  state = {
+    photos: [],
+    listFound: false,
+  };
+
+  componentWillReceiveProps(nextProps) {
+    if (this.props.photo !== nextProps.photo) {
+      if (nextProps.photo.length > 0) {
+        // fetch 4 photos from random starting index point
+        if (nextProps.photo.length > 4) {
```

Comment:
Since >4 always >0, why just mix the two if statements to one?

Reply:
because, if it is greater than 0 but less then 4 it will go into the else {           this.fetchPhoto(nextProps.photo[0]);
          this.fetchPhoto(nextProps.photo[1]);
          this.fetchPhoto(nextProps.photo[2]);
          this.fetchPhoto(nextProps.photo[3]);
}

Reply:
Another choice is 

if (nextProps.photo.length > 4) {
...
} else if (nextProps.photo.length > 0) {
...
}

or good practise is 
put comment to tell others what is the scope of the else?

Reply:
A good practice is not a comment, but a readable/understandable `if` condition at first glance

Reply:
my pseudocode will be like:
IF there are any photos THEN
     IF there are more than four photos THEN
            do A
     ELSE
            do B

your suggestion pseudocode would be like:
IF there are more than four photos THEN
     do A
ELSE IF there are more than zero photos THEN
     do B

Their output are the same, but they are checking different things at first?

Reply:
Alright Alvin